woocommerce/woocommerce
# Build Scripts: Make build and build:zip more intuitive

### All Submissions:






### Changes proposed in this Pull Request:

### Note to Reviewer
This PR references [a branch in our build Github Action](https://github.com/woocommerce/action-build/pull/10). The Action PR should be merged first, then the reference updated to `trunk` in this WooCommerce PR, and finally merged.


In an effort to standardize build scripts across the monorepo, its required that build steps only build and avoid also creating a zip.

This PR changes Core's `build` to only build core while `build:zip` builds Core but also creates a zip. This change made `build:core` redundant, so its been removed. A `build_step` config option was added so that WooRelease can still use the custom `build:zip` script instead of defaulting to `build`.

Partially fixes https://github.com/woocommerce/woocommerce/issues/32654

### How to test the changes in this Pull Request:

1. TBD: Confirm WooRelease is using the `build:zip` script (This is currently not possible, see https://github.com/woocommerce/woorelease/issues/361)
```
php woorelease.php simulate https://github.com/woocommerce/woocommerce/tree/fix/core-build-scripts --force
```
Check logs for presence of echoed text found in the [build zip script](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/bin/build-zip.sh).

Since testing by CLI is not currently possible, instead ensure `build_step` was correctly added to `package.json` configs.

3. Try building `woocommerce` without creating a zip `pnpm nx build woocommerce`
4.  Try building `woocommerce` while also creating a zip `pnpm nx build:zip woocommerce`

### Other information:




### FOR PR REVIEWER ONLY:
